File: evd_ros_backend/evd_ros_core/src/evd_script/program/machine_operations/machine_stop.py
'''
Commands machine to stop running a routine. Actual implemenation subject
to application engineer.

This might not do anything if the process is already done (and that is fine).
Of course stopping a process early might be problematic so care should be taken
when implementing this in an application.
'''

from .machine_primitive import MachinePrimitive


class MachineStop(MachinePrimitive):

    '''
    Data structure methods
    '''

    @classmethod
    def type_string(cls, trailing_delim=True):
        return 'machine-stop' + ('.' if trailing_delim else '')

    @classmethod
    def full_type_string(cls):
        return MachinePrimitive.full_type_string() + cls.type_string()

    def __init__(self, machineUuid=None, type='', name='', uuid=None, parent=None, append_type=True):
        super(MachineStop,self).__init__(
            machineUuid=machineUuid,
            type=MachineStop.type_string() + type if append_type else type,
            name=name,
            uuid=uuid,
            parent=parent,
            append_type=append_type)

    '''
    Execution methods
    '''

    def symbolic_execution(self, hooks):
        pass

    def realtime_execution(self, hooks):
        pass

File: evd_authoring_app/src/model/evdScript/program/machineOperations/machineStop.js
import { MachinePrimitive } from './machinePrimitive';

export class MachineStop extends MachinePrimitive {

    /*
    * Data structure methods
    */

    static typeString() {
        return 'machine-stop';
    }

    static fullTypeString() {
        return MachinePrimitive.fullTypeString() + MachineStop.typeString();
    }

    constructor(machineUuid=null, type='', name='', uuid=null, parent=null, appendType=true) {
        super(
            machineUuid,
            (appendType) ? 'machine-stop'+type : type,
            name,
            uuid,
            parent,
            appendType
        );
    }

    static fromDict(dct) {
        return new MachineStop(
            dct.machine_uuid,
            dct.type,
            dct.name,
            dct.uuid,
            null,
            false
        );
    }

    static BlocklyToolbox() {
        return { type: 'machine_stop' };
    }

    static BlocklyBlock() {
        return { key: 'machine_stop', data: {
            init: function() {
                this.appendDummyInput()
                    .appendField("Machine Stop");
                this.appendValueInput("machine")
                    .setCheck("machine")
                    .appendField("Machine");
                this.setPreviousStatement(true, null);
                this.setNextStatement(true, null);
                this.setColour(120);
                this.setTooltip("machine-stop");
                this.setHelpUrl("machine-stop");
            }
        }};
    }
}